tests/qml/tst_mainWindowGleanEvents.qml

```diff
@@ -0,0 +1,64 @@
+import QtQuick 2.3
+import QtTest 1.0
+
+import TestHelper 1.0
+import org.mozilla.Glean 0.24
+import ui 0.1
+
+Item {
+    Main {
+        id: mainTestGlean
+    }
+    
+    TestCase {
+        name: "MainTestsGleanEvents"
+
+        /**
+         * Taken from: https://github.com/mozilla/glean.js/blob/main/samples/qt/src/Tests/tst_maintests.qml
+         * 
+         * This function will block until a given promise is fulfilled.
+         *
+         * Note: This needs to be inside a TestCase element,
+         * because `tryVerify` is a test only API.
+         */
+        function awaitOn(promise) {
+            // Whether or not the promise fulfilled, doesn't matter if resolved or rejected.
+            let promiseFulfilled = false;
+            // Whether or not the promise resolved succesfully.
+            let promiseResolved = false;
+            // The result of the resolved promise;
+            let promiseData = undefined;
+
+            promise
+                .then(data => {
+                    promiseResolved = true;
+                    promiseFulfilled = true;
+                    promiseData = data;
+                })
+                .catch(() => promiseFulfilled = true);
+
+            // This will block waiting for the promise to fulfill.
+            tryVerify(
+                () => promiseFulfilled,
+                5000,
+                `Timed out while waiting for promise to fulfill.`
+            );
+
+            return {
+                resolved: promiseResolved,
+                data: promiseData
+            }
+        }
+
+        function test_onRecordGleanEventRecordsAppropriateSample() {
+            TestHelper.triggerRecordGleanEvent("authenticationAborted")
+            const { data: authenticationAbortedData } = awaitOn(Sample.authenticationAborted.testGetValue());
```

Comment:
The `test*` APIs can only be called when Glean is in "test mode". Glean is put in test mode after you call the `Glean.testResetGlean` API. It is important that this API is called before the app code calls `Glean.initialize`, this makes for a not so nice code but I have a bug to try to make it nicer. This is how I do that on the QML sample tests https://github.com/mozilla/glean.js/blob/main/samples/qt/src/Tests/tst_maintests.qml#L76-L85 :)

Reply:
Thank you for taking the time to explain this!